### PR TITLE
Desktop Leaderboard

### PR DESCRIPTION
**Pull request by simmiggrwl**
implemented web ui for leaderboard 
and fixed a minor bug in mobile implementation of the same

### CODE REVIEW

File: lib/pages/leaderboard_page/components/leaderboard_page_builder_web.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      _LeaderboardPageBuilderWebState();
+}
+
+class _LeaderboardPageBuilderWebState extends State<LeaderboardPageBuilderWeb> {
+  @override
+  initState() {
+    super.initState();
+    context
+        .read<LeaderboardCubit>()
+        .getLeaderboard(1, 10, widget.leaderboardType);
```
Comment (siva2204) [inline lib/pages/leaderboard_page/components/leaderboard_page_builder_web.dart:24]: idt we need this, you're calling this again in lib/pages/leaderboard_page/components/leaderboard_table_web.dart

File: lib/pages/leaderboard_page/leaderboard_page.dart [hunk 2]
```diff
@@ -57,85 +48,213 @@ class _LeaderboardPageState extends State<LeaderboardPage> {
...
+                    ],
+                  ),
+                  SizedBox(
+                    height: MediaQuery.of(context).size.height,
+                    child: TabBarView(
+                        physics: const BouncingScrollPhysics(
+                            parent: AlwaysScrollableScrollPhysics()),
+                        children: [
+                          BlocProvider(
+                            create: (context) => LeaderboardCubit(),
```
Comment (siva2204) [inline lib/pages/leaderboard_page/leaderboard_page.dart:241]: you are creating a cubit here, and creating again the same instance of cubit in the child level, i think creating any one of them will do.

same goes for all other 4 places also

File: lib/pages/leaderboard_page/components/leaderboard_table_web.dart [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+    super.initState();
+    context
+        .read<LeaderboardCubit>()
+        .getLeaderboard(1, 10, widget.leaderboardType);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    int x = 10;
+    int y = 1;
```
Comment (siva2204) [inline lib/pages/leaderboard_page/components/leaderboard_table_web.dart:30]: meaningful naming are highly appreciated

File: lib/pages/leaderboard_page/components/leaderboard_table_web.dart [hunk 4]
```diff
@@ -0,0 +1,664 @@
...
+                                            ' participants. ',
+                                            style: TextStyle(fontSize: 18),
+                                          ),
+                                        ],
+                                      )))
+                            ],
+                          ),
+                        ),
+                      ],
+                    );
```
Comment (siva2204) [inline lib/pages/leaderboard_page/components/leaderboard_table_web.dart:413]: this is code part is repeated twice, under `DailyLeaderboardSuccess` and `OverallLeaderboardSuccess`

can create a widget and pass leaderboardType as parameter ?

File: lib/pages/leaderboard_page/components/leaderboard_table_web.dart [hunk 5]
```diff
@@ -0,0 +1,664 @@
...
+                  color: background2),
+              child: Padding(
+                padding:
+                    const EdgeInsets.symmetric(vertical: 10, horizontal: 20),
+                child: ConstrainedBox(
+                  constraints: const BoxConstraints(
+                    minWidth: 120.0,
+                    maxWidth: 120.0,
+                  ),
+                  child: Image(image: AssetImage(imagePath)),
```
Comment (siva2204) [inline lib/pages/leaderboard_page/components/leaderboard_table_web.dart:496]:
```suggestion
                  child: rank <= 3 ? Image(image: AssetImage(imagePath)) : null,
```

Comment (siva2204) [reply lib/pages/leaderboard_page/components/leaderboard_table_web.dart:496]: can remove this if else part with the above condition ig

**Review by siva2204 — CHANGES_REQUESTED** [review]

File: lib/pages/leaderboard_page/components/leaderboard_page_builder_web.dart [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  @override
+  State<LeaderboardPageBuilderWeb> createState() =>
+      _LeaderboardPageBuilderWebState();
+}
+
+class _LeaderboardPageBuilderWebState extends State<LeaderboardPageBuilderWeb> {
+  @override
+  initState() {
+    super.initState();
+  }
```
Comment (dhruv1294) [inline lib/pages/leaderboard_page/components/leaderboard_page_builder_web.dart:22]: remove this initstate if not using

**Review by dhruv1294 — APPROVED** [review]

**Review by siva2204 — APPROVED** [review]

**Pull request merged**